### ISSUE TITLE
Odd dashpanel behavior using intellicenter and ultratemp heat cool pump

### ISSUE THREAD

**Issue by lockeandev** [rstrouse/nodejs-poolController-dashPanel#35]
I have an intellicenter i8PS with an ultratemp heat/cool heat pump. when I have the unit set to heat, the cooling icon shows in the dash panel and the dashpanel says "cooling".  also, the radio button only show "off" and blank options. I can't select the blank option and once the off option is selected, the only way to get the heater back "on" is to do so from the intellicenter panel itself, or from the pentair intellicenter app.

[replay.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/7029317/replay.zip)
<img width="525" alt="2021-08-22 22_45_15-Pool Control Dashboard" src="https://user-images.githubusercontent.com/11232490/130394269-9733f111-a640-4fd4-8278-10629ec67414.png">

**Comment by rstrouse** [comment]
Wow that is very odd.  I'll have a look at your replay.  There are other users that have Ultratemp and we worked through some issues with the way it was reporting.  Fortunately, I documented what we found with that one but there were a number of bytes that were never populated on that one.  Perhaps you may help us fill in some gaps with the settings.

**Comment by rstrouse** [comment]
Can you tell me what expansions you have installed on your i8PS?

It looks like we are detecting an i8PS personality card, a valve expansion module, and what was originally identified as an iChlor multiplexer card.  However, I didn't think they had even started shipping that card yet.  Do you have one of these for multiple chlorinators?

I would be really interested in a picture of the card installations in your IntelliCenter if that is possible.  This would be of the flip down plate at the top of the IntelliCenter control box.

**Comment by lockeandev** [comment]
I have an i8PS personality card (although I don't have a spa), a valve expansion module, and the third card is a Pool Cover expansion Module (PN 522039Z). I don't have an iChlor Multiplexer card.

I have an air and water temp sensor, but just have a 10k resistor in the solar sensor location as I don't have solar.

I can take a picture later tonight.

**Comment by rstrouse** [comment]
Ok so I the first thing you need to do is update the firmware on your IntelliCenter.  Perhaps because you do not have a spa the bugs may not be so glaring but there were so many on 1.040.  The one thing that will end up driving you nuts is what it does with heaters and setpoints related to the schedules.   It will randomly change these values on you (actually it is not random but it feels like it is as each time it triggers a schedule the solar setpoints get switched out for the heater setpoints).  There are a ton of other items but that was the thing that caused me the greatest grief.

1.047 is much more stable.  Quite a few of the settings on 1.040 didn't work correctly so please update your system.

It looks like we have the bits flipped between the mux card and the cover controller so thank you for that.  I have also found that we probably need another bit of code for the solar sharing on the heater with the Ultratemp because of the requirement for the solar temp probe when running the Ultratemp without a solar or gas backup.  The previous user with Ultratemp had solar as well so that blank item would have read Solar Preferred.

I'll push out another update in the next 24 hours after I have examined the replay completely.  Thanks again.

EDIT:
Here is a link to the firmware.
https://www.pentair.com/en-us/education-support/residential/product-support/pentair-pool-and-spa-software-downloads/intellicenter-download.html

**Comment by rstrouse** [comment]
I have verified a few things with the replay.  Your byte output from the heater is correct but IntelliCenter is ignoring the cooling bytes and treating the UltraTemp as a regular old heat pump which has much of the functionality but not all of it.  This is another area of the 1.040 firmware that had fixes applied when Pentair released 1.047.  

Upgrading your firmware should make dashPanel and njsPC recognize the settings on the heater.  I will have an update shortly to identify your cover controller correctly tonight but we don't surface any of the cover information yet in dashPanel.  We were waiting until we ran into somebody that has one installed to see what kind of interface is needed.

**Comment by lockeandev** [comment]
Looks like the firmware update fixed the cooling and heating indicators on dash panel. The heat mode still says "off" when the heater or cooler is definitely on, and the Solar preferred option is still blank. Not sure what is supposed to be listed on the heat mode but heater status appears to update correctly now.

I thought I had updated the firmware before this and when I went out to do it just now, the wireless update said I was up to date. I had to do the update from USB.
![Screenshot_20210823-181754-01](https://user-images.githubusercontent.com/11232490/130536382-d5635f99-06b7-4f23-830e-3d373b1a2286.jpeg)
![Screenshot_20210823-182156-01](https://user-images.githubusercontent.com/11232490/130536393-d19b75b2-2ca5-47d7-bb5b-c4e10eebf64d.jpeg)

**Comment by rstrouse** [comment]
I will post an update tomorrow that fixes this issue.  There is a conflict with the heatpump settings.

**Comment by rstrouse** [comment]
I pushed up a couple of updates.  I would like you to update both njsPC and dashPanel.  Then capture another replay and post it here.

**Comment by lockeandev** [comment]
I know how to install both but is there a best practice for updating?  I've just been deleting the git cloned folders in the home directory and recloning and reinstalling via npm install

**Comment by rstrouse** [comment]
Navigate to each directory and type
`git pull`
`npm i`

Do this for each directory.

**Comment by lockeandev** [comment]
Done, It looks like it worked for the most part. when I first loaded the dashpanel it said "Solar Only" for the heat mode. once I made changes to the thermostat settings it updated. Although now, my 5 group circuits have disappear from the features.  They briefly reappeared during the config reload, but then disappeared when the config reload was complete. If I go into the circuits and then save a group circuit, it reappears in the features in the dashpanel. I didn't capture the process in the first replay tonight ([replay2.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/7043659/replay2.zip)) 
So I did a 2nd one tonight ([replay3.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/7043661/replay3.zip))

**Comment by rstrouse** [comment]
Thanks, I will have a look.  I have a suspicion regarding the group circuits.  

The solar only display is because IntelliCenter was still sending information regarding the configuration.   njsPC was still waiting on the heater settings before it can apply the correct terminology to the output for dashPanel to display.  Heaters configurations are some of the last equipment items sent prior to moving on to the schedules.

Are all the options for heat modes and status' now correct for your configuration?

**Comment by rstrouse** [comment]
Well, my suspicions were wrong regarding the Circuit Groups.

What browser are you using?  Please perform a hard refresh of the dashPanel browser.  
To do this:
In Windows:  Hold down the ctrl key + F5.
On Mac (anything but Safari): shift + command + R
On Mac using Safari: It is stupid in that you need to go into the options and clear the cache for the dashPanel URL.  This is true in IPad as well.

**Comment by lockeandev** [comment]
I only have the two options, Off and Ultra temp only, and those are now correct. Thanks.

I was using chrome, both on windows and android. Now that I have saved the circuit groups they all show up correctly. I will have to try to reproduce the behavior when I get home tonight. If the behavior persists after a hard refresh, I can open a new issue if you prefer.

**Comment by rstrouse** [comment]
One of the things that may have not been set is the Show As Feature checkbox.  If it isn't checked then dashPanel will not display it on the features section.

**Comment by rstrouse** [comment]
I am going to close this.  Open another issue if you have additional problems.
